### PR TITLE
[Feature] Add new string template functions

### PR DESCRIPTION
**Pull request by SirSova**
List of new funcs:
- lower
- upper
- camelcase
- snakecase
- kebabcase
- firstLower
- firstUpper

change the dependency for strings ops "iancoleman/strcase" -> "huandu/xstrings"

deprecate the custom "InterfaceName" template variables

Description
-------------

- Fixes # (https://github.com/vektra/mockery/issues/702)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

Version of Golang used when building/testing:
---------------------------------------------

- [ ] 1.11
- [ ] 1.12
- [ ] 1.13
- [ ] 1.14
- [ ] 1.15
- [ ] 1.16
- [ ] 1.17
- [ ] 1.18
- [ ] 1.19
- [x] 1.20

How Has This Been Tested?
---------------------------

Unit tests. Build and run mockery with new functions applied to a template locally.

Checklist
-----------

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by SirSova** [comment]
@LandonTClipp please review the PR, cuz I can't edit **Reviewers** section.

File: pkg/outputter.go [hunk 1]
```diff
@@ -215,9 +222,9 @@ func parseConfigTemplates(ctx context.Context, c *config.Config, iface *Interfac
 		InterfaceDir:            filepath.Dir(iface.FileName),
 		InterfaceDirRelative:    interfaceDirRelative,
 		InterfaceName:           iface.Name,
-		InterfaceNameCamel:      strcase.ToCamel(iface.Name),
-		InterfaceNameLowerCamel: strcase.ToLowerCamel(iface.Name),
-		InterfaceNameSnake:      strcase.ToSnake(iface.Name),
+		InterfaceNameCamel:      xstrings.ToCamelCase(iface.Name),
+		InterfaceNameLowerCamel: xstrings.FirstRuneToLower(xstrings.ToCamelCase(iface.Name)),
+		InterfaceNameSnake:      xstrings.ToSnakeCase(iface.Name),
```
Comment (LandonTClipp) [inline pkg/outputter.go:227]: Does this have any behavior differences or are they strictly equivalent? I'm inclined to keep the old package populating these variables and just remove them in v3.

Comment (SirSova) [reply pkg/outputter.go:227]: as far as I see, they behave absolutely the same.
But since they are deprecated I guess it's okay to keep code unchanged.

Comment (SirSova) [reply pkg/outputter.go]: Done

**Review by LandonTClipp — APPROVED** [review]
Great!

**Review by LandonTClipp — APPROVED** [review]
.

**Pull request merged**